lib/elinux_build_target.dart

```diff
@@ -351,24 +351,23 @@ class NativeBundle {
 
     // Run the native build.
     final String cmakeBuildType = buildMode.isPrecompiled ? 'Release' : 'Debug';
-    final String targetArch = buildInfo.targetArch;
-    final String hostArch = _getCurrentHostPlatformArchName();
+    final String targetArch = buildInfo.targetArch == 'arm64' ? 'aarch64' : 'x86_64';
     final String targetCompilerTriple = buildInfo.targetCompilerTriple;
     final String targetSysroot = buildInfo.targetSysroot;
     final String systemIncludeDirectories = buildInfo.systemIncludeDirectories;
-    final bool needCrossBuild = targetArch != hostArch;
-    final bool needCrossBuildOptionsForArm64 = targetArch == 'arm64';
+    final bool needCrossBuild = targetSysroot != '/';
```

Comment:
This condition would not be considered cross builds if only the architecture differs. (e.g. triple) What you want to do is make sysroot and target CPU independent, right?

Reply:
I believe is not possible to have the same sysroot with different architecture, because sysroot have its own architecture. For example, if `/` is x64, if I build arm64 app, it is impossible to use this sysroot.

Reply:
I agree 50%. There are some build causes. For instance, builds with `g++-aarch64-linux-gnu` and `Yocto SDK (pnc builds)`.

Reply:
Updated, please take a look again, thanks.